pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -51,7 +51,8 @@ func (p Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionIn
 	// ServiceAccount for masters
 	// Only create ServiceAccount for masters if a shared VPC install is not being done
 	if len(in.InstallConfig.Config.Platform.GCP.NetworkProjectID) == 0 ||
-		in.InstallConfig.Config.ControlPlane.Platform.GCP.ServiceAccount == "" {
+		(in.InstallConfig.Config.ControlPlane.Platform.GCP != nil &&
```

Comment:
I think this will segfault if you don't define `controlPlane` stanza in the install-config.yaml.